src/libraries/System.Threading.RateLimiting/src/System/Threading/RateLimiting/ChainedPartitionedRateLimiter.cs

```diff
@@ -23,21 +23,36 @@ public ChainedPartitionedRateLimiter(PartitionedRateLimiter<TResource>[] limiter
             _limiters = limiters;
         }
 
-        public override int GetAvailablePermits(TResource resource)
+        public override RateLimiterStatistics? GetStatistics(TResource resource)
         {
             ThrowIfDisposed();
-            int lowestPermitCount = int.MaxValue;
+            long lowestAvailablePermits = long.MaxValue;
+            long lowestQueuedCount = long.MaxValue;
+            long totalFailedLeases = 0;
+            long totalSuccessfulLeases = 0;
             foreach (PartitionedRateLimiter<TResource> limiter in _limiters)
             {
-                int permitCount = limiter.GetAvailablePermits(resource);
-
-                if (permitCount < lowestPermitCount)
+                if (limiter.GetStatistics(resource) is { } statistics)
                 {
-                    lowestPermitCount = permitCount;
+                    if (statistics.CurrentAvailablePermits < lowestAvailablePermits)
+                    {
+                        lowestAvailablePermits = statistics.CurrentAvailablePermits;
+                    }
+                    if (statistics.CurrentQueuedCount < lowestQueuedCount)
+                    {
+                        lowestQueuedCount = statistics.CurrentQueuedCount;
+                    }
+                    totalFailedLeases += statistics.TotalFailedLeases;
+                    totalSuccessfulLeases += statistics.TotalSuccessfulLeases;
```

Comment:
On the other hand, I'm not sure it makes sense for these totals to be summed. This ends up double counting successful leases.

It probably more makes sense for `ChainedPartitionedRateLimiter` to do its own accounting of failed and successful leases, and only count fully acquiring the chain as success.

Thinking about it more, we could probably account for the `CurrentQueuedCount` at this layer just by tracking how many `AcquireAsync` calls are ongoing. `CurrentAvailablePermits` might be the only thing that cannot be independently calculated by `ChainedPartitionedRateLimiter` without looking at the inner limiter's statistics.

Reply:
> at this layer just by tracking how many `AcquireAsync` calls are ongoing.

The issue here is that it's tracked per resource, which means `ChainedPartitionedRateLimiter` would need to keep per-resource storage, which then means it needs a way to trim that storage etc.

Reply:
We should be able to look at the innermost limiter's successful leases.